src/sd.rs

```diff
@@ -0,0 +1,301 @@
+use core::borrow::Borrow;
+
+use crate::spi::SpiDriver;
+use crate::sys::*;
+
+#[cfg(esp_idf_soc_sdmmc_host_supported)]
+use mmc::SdMmcHostDriver;
+use spi::SdSpiHostDriver;
+
+#[cfg(esp_idf_soc_sdmmc_host_supported)]
+pub mod mmc;
+pub mod spi;
+
+const _SDMMC_HOST_FLAG_SPI: u32 = 1 << 3;
+const _SDMMC_HOST_FLAG_DDR: u32 = 1 << 4;
+const _SDMMC_HOST_FLAG_DEINIT_ARG: u32 = 1 << 5;
+
+pub type SdCardConfiguration = config::Configuration;
+
+pub mod config {
+    #[cfg(not(any(
+        esp_idf_version_major = "4",
+        all(esp_idf_version_major = "5", esp_idf_version_minor = "0"),
+        all(esp_idf_version_major = "5", esp_idf_version_minor = "1"),
+    )))] // For ESP-IDF v5.2 and later
+    use crate::sys::*;
+
+    /// (SD-MMC only): Input delay phase
+    #[cfg(not(any(
+        esp_idf_version_major = "4",
+        all(esp_idf_version_major = "5", esp_idf_version_minor = "0"),
+        all(esp_idf_version_major = "5", esp_idf_version_minor = "1"),
+    )))] // For ESP-IDF v5.2 and later
+    #[non_exhaustive]
+    #[derive(Debug, Copy, Clone, Eq, PartialEq)]
+    pub enum DelayPhase {
+        Phase0 = sdmmc_delay_phase_t_SDMMC_DELAY_PHASE_0 as isize,
+        Phase1 = sdmmc_delay_phase_t_SDMMC_DELAY_PHASE_1 as isize,
+        Phase2 = sdmmc_delay_phase_t_SDMMC_DELAY_PHASE_2 as isize,
+        Phase3 = sdmmc_delay_phase_t_SDMMC_DELAY_PHASE_3 as isize,
+    }
+
+    #[cfg(not(any(
+        esp_idf_version_major = "4",
+        all(esp_idf_version_major = "5", esp_idf_version_minor = "0"),
+        all(esp_idf_version_major = "5", esp_idf_version_minor = "1"),
+    )))] // For ESP-IDF v4.x, v5.0, and v5.1
```

Comment:
not(any( and //For ESP-IDF v4.x is a contradiction

Reply:
Copy pasted code from the original PR. I guess the author meant "NOT any..." in the comment. Will fix.